### PR TITLE
update ember-cli project and dependencies

### PR DESCRIPTION
**Pull request by miguelcobain**
This PR:
- updates the project to the latest ember/ember-cli versions (used ember-cli-update)
- updates all its dependencies, including auth0/auth0-lock
- updates the code styling of the dummy app to use native classes
- stops using getWithDefault and getProperties which is now deprecated
- no longer uses `get()` in favor of using direct property access
- a few test updates and simplifications

Basically moves this addon a bit more into 2020.

### CODE REVIEW

**Comment by miguelcobain** [comment]
For some reason, it didn't seem like the tests success was reported, but the build did succeed on travis: https://travis-ci.org/github/auth0-community/ember-simple-auth-auth0/builds/576890741

@XaserAcheron can we get this merged? I'm not sure anymore who became this repository's maintainer. There are no updates in more than a year.

**Comment by XaserAcheron** [comment]
Hey! Dang, it's been a while...

I guess the keys to this repo never got handed over to someone officially :( -- I'm not in a position to be able to support this addon these days (haven't written a line of JS, much less Ember, in over a year now), though that means I trust your changes way more than I trust anything of mine. :P

May as well merge this, though yeah, it'd still be best if we could find someone willing to "keep the lights on," as it were.
